packages/abi-coder/src/index.ts

```diff
@@ -1,21 +1,17 @@
-export * from './coders/abstract-coder';
-export { default as Coder } from './coders/abstract-coder';
-export { default as ArrayCoder } from './coders/array';
-export { default as B256Coder } from './coders/b256';
-export { default as B512Coder } from './coders/b512';
-export { default as BooleanCoder } from './coders/boolean';
-export { default as ByteCoder } from './coders/byte';
-export { default as EnumCoder } from './coders/enum';
-export { default as NumberCoder } from './coders/number';
-export { default as StringCoder } from './coders/string';
-export { default as StructCoder } from './coders/struct';
-export { default as TupleCoder } from './coders/tuple';
-export { default as U64Coder } from './coders/u64';
-export { default as VecCoder } from './coders/vec';
-export * from './utilities';
-export * from './fragments/fragment';
-export { default as FunctionFragment } from './fragments/function-fragment';
-export { default as Interface } from './interface';
-export { default as AbiCoder } from './abi-coder';
-export * from './json-abi';
-export * from './constants';
+export { Coder, InputValue, DecodedValue } from './coders/abstract-coder';
+export { ArrayCoder } from './coders/array';
+export { B256Coder } from './coders/b256';
+export { B512Coder } from './coders/b512';
+export { BooleanCoder } from './coders/boolean';
+export { ByteCoder } from './coders/byte';
+export { EnumCoder } from './coders/enum';
+export { NumberCoder } from './coders/number';
+export { StringCoder } from './coders/string';
+export { StructCoder } from './coders/struct';
+export { TupleCoder } from './coders/tuple';
+export { U64Coder } from './coders/u64';
+export { VecCoder } from './coders/vec';
+export type { FunctionFragment } from './function-fragment';
```

Comment:
Because `Interface` is the entry point through which interaction with `FunctionFragment`s happens, only the type (and not the class) of `FunctionFragment` is exported for the needs of e.g. building a `FunctionInvocationScope`, as it's tied to the `FunctionFragment` specifically.